code/modules/hydroponics/grown/misc.dm

```diff
@@ -28,6 +28,40 @@
 
 	parent.update_tray()
 
+// Corpse flower
+/obj/item/seeds/starthistle/corpse_flower
+	name = "pack of corpse flower seeds"
+	desc = "A species of plant that emits a horrible odor."
+	icon_state = "seed-corpse-flower"
+	species = "corpse-flower"
+	plantname = "Corpse flower"
+	production = 2
+	growing_icon = 'icons/obj/hydroponics/growing_flowers.dmi'
+	genes = list()
+	mutatelist = list()
+
+/obj/item/seeds/starthistle/corpse_flower/pre_attack(obj/machinery/hydroponics/I)
+	if(ishydroponics(I))
+		if(!I.myseed)
+			START_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/seeds/starthistle/corpse_flower/process()
+	var/obj/machinery/hydroponics/parent = loc
+	if(parent.age < maturation) // Start a little before it blooms
+		return FALSE
+
+	var/turf/open/T = get_turf(parent)
+	if(T.return_air().return_pressure() > (ONE_ATMOSPHERE + potency/10 + 10)) // clouds can begin showing at around 50-60 potency in standard atmos
+		return TRUE
+
+	var/datum/gas_mixture/stank = new
+	ADD_GAS(/datum/gas/miasma, stank.gases)
+	stank.gases[/datum/gas/miasma][MOLES] = (yield + 6)*7*MIASMA_CORPSE_MOLES // this process is only being called about 1/7 as much as corpses
+	stank.temperature = T20C // without this the room would eventually freeze and miasma mining would be easier
```

Comment:
The temperature thing should be a problem elsewhere in the code, can I fix it in this PR?